test/cuda/cufinufft1dspreadinterponly_test.cu

```diff
@@ -0,0 +1,215 @@
+#include <cmath>
+#include <complex>
+#include <iomanip>
+#include <iostream>
+#include <random>
+
+#include <cufinufft.h>
+
+#include <cufinufft/contrib/helper_cuda.h>
+#include <cufinufft/impl.h>
+#include <cufinufft/utils.h>
+
+#include <thrust/complex.h>
+#include <thrust/device_vector.h>
+#include <thrust/host_vector.h>
+
+using cufinufft::utils::infnorm;
+
+template<typename T>
+int run_test(int method, int type, int N1, int M, T tol, T checktol, int iflag,
+             double upsampfac) {
+  // tol and upsamplefac are used to determine the kernel
+  if (type < 1 || type > 3) {
```

Comment:
delete this since type is ignored (really should be removed from args of run_test()).